src/pocketmine/Player.php

```diff
@@ -2520,7 +2542,15 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 							}
 						}
 
-						$this->setUsingItem(true);
+						if($this->startAction === -1){
+							$this->setUsingItem(true);
+							return true;
+						}
+
+						$this->setUsingItem(false);
+						if($item->onUse($this)){
```

Comment:
This change creates a behavioural break for food consumption. The events and benefits are supposed to apply after the item is finished being eaten (during the CONSUME transaction type), not when the player just started using the item.